### ISSUE TITLE
Better DataSet module naming/alignment in core

### ISSUE THREAD

**Issue by MatthewFlamm** [pyvista/pyvista#4535]
## Issue
Now that `UniformGrid` - > `ImageData` from #4518 and a big module reorg in #4486, I think we should evaluate whether we can clean up the module names for `DataSet`s.  This has always been a pet peeve of mine.  Today, excluding deprecated classes, we have:

In `pyvista.core.pointset`:
 - `PointSet`
 - `PolyData`
 - `PointGrid`
 - `UnstructuredGrid`
 - `StructuredGrid`
 - `ExplicitStructuredGrid`

In `pyvista.core.grid`:
 - `RectilinearGrid`
 - `ImageData`

Note that there are 4 `*Grids` in `pyvista.core.pointset` (2/3 of the `DataSet`s in that module) and only 1 `*Grid` in `pyvista.core.grid` (1/2 of the `DataSet`s in that module).

## Proposal:

I propose following vtk's inheritance tree for organization:

![image](https://github.com/pyvista/pyvista/assets/39341281/86130535-95aa-4ea7-b11b-5d12d39fe6f0)

This means that `pointset` remains unchanged.  But then `ImageData` and `RectilinearGrid` would each get their own module (`image_data` and `rectilinear_grid`?).  Or we come up with another catchall name for a non-pointset DataSet.

One complication is that `RectilinearGrid` and `ImageData` both inherit from `Grid` class.  From a practical future looking standpoint, there are only a few methods in common and this could easily become a private class for convenience, or the methods could be duplicated in each class.

### Other implications

Tests also sorely need to be reorganized even if we don't implement this proposal.  But, this will fall into line once a decision is made.  For example, `UnstructuredGrid` tests are largely in `test_grid.py` rather than `test_pointset.py` or its own test module.  I think this demonstrates the confusion around the naming of the module in `core`.

**Comment by banesullivan** [comment]
+100 on the reorganization, and I'm happy to take this on rather immediately.

re tests... this indeed needs a lot of reorganization. I'm leaning towards one test module (file) per class. I.e. `test_rectilinear_grid.py`

**Comment by MatthewFlamm** [comment]
> re tests... this indeed needs a lot of reorganization. I'm leaning towards one test module (file) per class. I.e. `test_rectilinear_grid.py`

This makes sense to me, but there may be some tests that parametrize across all the pointsets, so it might have to be broken up into common pointset tests and specific tests.

**Comment by akaszynski** [comment]
> I'm leaning towards one test module (file) per class

Agreed. I'd also like one test module for each filter class. We've just about combined all the filter classes into a single file and it makes it difficult to find the right test as well as isolate testing.
